api/base/permissions.py

```diff
@@ -76,3 +78,17 @@ def has_object_permission(self, request, view, obj):
             obj = self.get_object(request, view, obj)
             return super(Perm, self).has_object_permission(request, view, obj)
     return Perm
+
+
+class OsfBasePermission(permissions.BasePermission):
+
+    def has_permission(self, request, view):
+        if request.user.is_anonymous():
```

Comment:
Should this logic not be handled in the base/authentication/drf.py module? Seems the user should never get logged in if these conditions exist.